Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -181,6 +171,73 @@ theorem prod_trunc {M : Type*} [CommMonoid M] {a b : ℕ} (f : Fin (a + b) → M
   rw [prod_univ_add, Fintype.prod_eq_one _ hf, mul_one]
   rfl
 
+section Interval
+
+variable {M : Type*} [CommMonoid M] {n : ℕ} {v : Fin (n + 1) → M}
+
+@[to_additive]
+theorem prod_Ioi_zero : ∏ i ∈ Ioi 0, v i = ∏ j : Fin n, v j.succ := by
+  rw [Ioi_zero_eq_map, Finset.prod_map, val_succEmb]
+
+@[to_additive]
+theorem prod_Ioi_succ (i : Fin n) : ∏ j ∈ Ioi i.succ, v j = ∏ j ∈ Ioi i, v j.succ := by
+  rw [Ioi_succ, Finset.prod_map, val_succEmb]
+
+@[to_additive (attr := simp)]
+theorem prod_Ici_zero : ∏ i ∈ Ici 0, v i = ∏ j : Fin (n + 1), v j := by
+  rw [Ici_eq_Icc, ← bot_eq_zero, Icc_bot_top]
+
+@[to_additive]
+theorem prod_Ici_succ (i : Fin n) :
+    ∏ j ∈ Ici i.succ, v j = ∏ j ∈ Ici i, v j.succ := by
+  rw [Ici_succ, Finset.prod_map, val_succEmb]
+
+@[to_additive (attr := simp)]
+theorem prod_Iio_zero : ∏ i ∈ Iio 0, v i = 1 := rfl
+
+@[to_additive]
+theorem prod_Iio_succ (i : Fin n) :
+    ∏ j ∈ Iio i.succ, v j = (∏ j ∈ Iio i.castSucc, v j) * v i.castSucc := by
+  calc
+    _ = ∏ j ∈ Icc 0 i.castSucc, v j := prod_congr rfl (fun _ _ => rfl)
```

Comment:
These don't look that rfl; are there lemmas you can use instead?

Reply:
Note that the second `rfl` is actually trivial (of the form `v x = v x`), only the first is not (and this is due to how `Icc` is defined for `Fin`).

I can't find any lemma for the first `rfl`. Such a lemma would need to be particular to `Fin`, or any canonically ordered type.